### DISCUSSION TITLE
When using httpx's screenshot params, it creates a lot of zombie processes that can't be killed.

### DISCUSSION THREAD

**Discussion by monitor403** [projectdiscovery/httpx#1415] [Q&A]
When I use httpx's screenshot function to get website images in bulk, the process generates a lot of zombie processes, and after a large number of zombie processes have appeared, I can no longer create the httpx task, and report an error: **runtime: failed to create new OS thread (have 19 already; errno=11)**
![image](https://github.com/projectdiscovery/httpx/assets/45124775/32f57dd8-a3e1-4ef8-b6fe-5e43c9f7cbe8)

**Comment by MetzinAround** [comment]
Hi! Could you tell us some more information? 

- what's the full command?
- what OS are you using?
- what version of httpx are you running?

Thanks!

**Reply by monitor403** [reply]
Hi~ here is my information:
Full command: httpx -title -duc -tech-detect -no-stdin -status-code -fr -nf -json -ss -u http://example.com
OS Version: Linux 1b10610f65fd 6.1.0-kali5-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.12-1kali2 (2023-02-23) x86_64 GNU/Linux
httpx version: v1.3.4

**Reply by monitor403** [reply]
@MetzinAround In fact, I'm having this problem when I run the latest version of httpx on a debain machine to get a snapshot. It looks like httpx doesn't wait for the chrome process to finish, so the process is never released.

**Reply by monitor403** [reply]
@MetzinAround hello,has there been any progress on this discuss? Looking forward to your response.

**Reply by MetzinAround** [reply]
Hi @monitor403. Thanks for the ping! I have brought it to the OSS team and asked for some help.

**Reply by RamanaReddy0M** [reply]
The issue is being tracked here: https://github.com/projectdiscovery/httpx/issues/1425. Please follow it for updates. @monitor403

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
